builder/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM golang:1.18.1-bullseye as build-env
+
+WORKDIR /go/src/app
+
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+
+COPY cmd cmd
+COPY lib lib
+COPY Makefile .
+
+RUN make all
```

Comment:
For local development, one can use `make test` or `go test <etc>` instead of using this.

What I want the answer to be is "there is no need to re-run tests unless the input (e.g. lib, cmd, Makefile) changes, as the test output is a deterministic function of the input". But this isn't strictly true, I've already written a heavyweight unit test that might pass or fail based on the whims of goroutine scheduling if the implementation is defective.

What kind of scenarios did you have in mind where we'd want to re-run the tests?

another possible improvement could be to generate test reports and extract them out of the build container as a build artefact.

Reply:
i've created an issue to track this, i propose not trying to fix this now: https://github.com/fcostin/tcplb/issues/10

but quite interested to hear your thoughts, if you have time.

please re open if you feel strongly that this needs to be fixed before we proceed

Reply:
I think what got me here is that I can't quite figure out the purpose for the docker image. No objections is keeping it as-is, though.

Reply:
the intent was to define a fairly controlled way to build and test the application, that is independent of people's dev environments or the build environment offered by CI.